clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/exceptions/SpinnakerNetworkException.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.exceptions;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
+import java.util.Optional;
+import lombok.Getter;
+import retrofit.RetrofitError;
+
+// todo(mneterval): move to kork-exceptions
+
+/** An exception that exposes the message and kind of a {@link RetrofitError}. */
+@Getter
+@NonnullByDefault
+public class SpinnakerNetworkException extends SpinnakerException {
+  private final String rawMessage;
+  private final RetrofitError.Kind kind;
```

Comment:
is it worth abstracting the `RetrofitError.Kind` out of here? Not that we would ever move away from retrofit (one hopes..?) but decoupling it here might be useful esp if a bunch of error handlers are going to be checking the `kind`